letsencrypt/client/tests/display/util_test.py

```diff
@@ -0,0 +1,349 @@
+
+    @mock.patch("letsencrypt.client.display.util.FileDisplay.input")
+    def test_checklist_miss_quit(self, mock_input):
+        mock_input.side_effect = [
+            (display_util.OK, "10"),
+            (display_util.CANCEL, "1")
+        ]
+        ret = self.displayer.checklist("msg", self.tags)
+        self.assertEqual(ret, (display_util.CANCEL, []))
+
+    def test_scrub_checklist_input_valid(self):
+        # pylint: disable=protected-access
+        indices = [
+            ["1"],
+            ["1", "2", "1"],
+            ["2", "3"],
+        ]
+        exp = [
+            set(["tag1"]),
+            set(["tag1", "tag2"]),
+            set(["tag2", "tag3"]),
+        ]
+        for i, list_ in enumerate(indices):
+            set_tags = set(
+                self.displayer._scrub_checklist_input(list_, self.tags))
+            self.assertEqual(set_tags, exp[i])
+
+    def test_scrub_checklist_input_invalid(self):
+        # pylint: disable=protected-access
+        indices = [
+            ["0"],
+            ["4"],
+            ["tag1"],
+            ["1", "tag1"],
+            ["2", "o"]
+        ]
+        for list_ in indices:
+            self.assertEqual(
+                self.displayer._scrub_checklist_input(list_, self.tags), [])
+
+    def test_print_menu(self):
+        # pylint: disable=protected-access
+        # This is purely cosmetic... just make sure there aren't any exceptions
+        self.displayer._print_menu("msg", self.choices)
+        self.displayer._print_menu("msg", self.tags)
+
+    def test_wrap_lines(self):
+        # pylint: disable=protected-access
+        msg = ("This is just a weak test\n"
+               "This function is only meant to be for easy viewing\n"
+               "Test a really really really really really really really really "
+               "really really really really really long line...")
+        text = self.displayer._wrap_lines(msg)
+
+        self.assertEqual(text.count(os.linesep), 3)
+
+    def test_get_valid_int_ans_valid(self):
+        # pylint: disable=protected-access
+        with mock.patch("__builtin__.raw_input", return_value="1"):
+            self.assertEqual(
+                self.displayer._get_valid_int_ans(1), (display_util.OK, 1))
+        ans = "2"
+        with mock.patch("__builtin__.raw_input", return_value=ans):
+            self.assertEqual(
+                self.displayer._get_valid_int_ans(3),
+                (display_util.OK, int(ans)))
+
+    def test_get_valid_int_ans_invalid(self):
+        # pylint: disable=protected-access
+        answers = [
+            ["0", "c"],
+            ["4", "one", "C"],
+            ["c"],
+        ]
+        for ans in answers:
+            with mock.patch("__builtin__.raw_input", side_effect=ans):
+                self.assertEqual(
+                    self.displayer._get_valid_int_ans(3),
+                    (display_util.CANCEL, -1))
+
+    # def test_visual(self):
```

Comment:
Well, unittests are therefore not good place to hide such manual testing suite... Why don't we move it to top-level `testing` directory (whatever name, but outside the `letsencrypt` Python package), uncommented, but with an appropriate note: "this allows you to quickly see examples of all the different display functions, but cannot be tested automatically through unittests", etc?